modules/engine/src/test/scala/com/gsk/kg/engine/optimizer/PushDownJoinOnValueSpec.scala

```diff
@@ -0,0 +1,594 @@
+                              Tuple2(
+                                StringVal.VARIABLE("?title"),
+                                StringVal.STRING("SPARQL Tutorial")
+                              ),
+                              Tuple2(
+                                StringVal.VARIABLE("?book"),
+                                StringVal.URIVAL(
+                                  "<http://example.org/book/book1>"
+                                )
+                              )
+                            )
+                          ),
+                          Row(
+                            ArrayBuffer(
+                              Tuple2(
+                                StringVal.VARIABLE("?title"),
+                                StringVal.STRING("The Semantic Web")
+                              ),
+                              Tuple2(
+                                StringVal.VARIABLE("?book"),
+                                StringVal.URIVAL(
+                                  "<http://example.org/book/book2>"
+                                )
+                              )
+                            )
+                          )
+                        )
+                      )
+                    ),
+                    joinR(
+                      quadR(
+                        StringVal.VARIABLE("?book"),
+                        StringVal.URIVAL("<http://example.org/ns#price>"),
+                        StringVal.VARIABLE("?price"),
+                        StringVal.GRAPH_VARIABLE :: Nil
+                      ),
+                      tableR(
+                        List(
+                          StringVal.VARIABLE("?book")
+                        ),
+                        List(
+                          Row(
+                            ArrayBuffer(
+                              Tuple2(
+                                StringVal.VARIABLE("?book"),
+                                StringVal.URIVAL(
+                                  "<http://example.org/book/book1>"
+                                )
+                              )
+                            )
+                          ),
+                          Row(
+                            ArrayBuffer(
+                              Tuple2(
+                                StringVal.VARIABLE("?book"),
+                                StringVal.URIVAL(
+                                  "<http://example.org/book/book2>"
+                                )
+                              )
+                            )
+                          )
+                        )
+                      )
+                    )
+                  )
+                )
+              case _ => fail
+            }
+          }
+          .getOrElse(fail)
+      }
+
+      "using a two variable in the clause, one with undefined values, one appearing in a single quad and the other in more than one quad" in {
```

Comment:
```suggestion
      "using two variables in the clause, one with undefined values, one appearing in a single quad and the other in more than one quad" in {
```
Again, this applies to all similar test descriptions.

Reply:
Fixed in two other places